### DISCUSSION TITLE
Usabilidade de um Quiz

### DISCUSSION THREAD

**Discussion by heliomsolivas** [frontendbr/forum#1021] [Perguntas e Respostas]
Pessoal, estou com o seguinte dilema aqui, tenho um quiz que a funcionalidade é a seguinte, se o usuário acertar ele continua respondendo (as perguntas aparecem na ordem randomica), se ele errar ele perde.

Porém se ele não acertar/errar ele consegue dar F5 ou fechar a janela, e entrar novamente no quiz, seria um meio do usuário "pular" uma pergunta do quiz.

Quais seriam as alternativas para lidar com isso? Creio que não é possível mostrar um overlay quando o usuário tentasse sair da página.

Para o usuário participar do Quiz ele tem que preencher um cadastro, fazer o login e inserir um código (fica salvo no back-end), depois disso ele começa o Quiz, a questão seriam alternativas de que ação tomar quando o usuário tentasse sair da página do Quiz.

**Comment by willycamargo** [comment]
Onde esta o código desse quiz? Quem gera essa ordem randômica para a lista de questões?

Se for no front, acredito que o ideal seria guardar já a "ordem" em que as questões foram sorteadas para o usuário atual, e também marcar quais questões já foram respondidas e qual é a atual, tudo no storage (loalstorage por exemplo).

Quando o usuário atualizar a página ele irá voltar na questão atual, que esta salva no storage.

Mas, falando em burlar, se o código estiver no front, é fácil ele burlar também, principalmente se não ofuscar esse storage.

**Comment by heliomsolivas** [comment]
@willycamargo é no back-end que é gerado a ordem randômica. 

Pensei no **onbeforeunload** porém não dá pra fazer muita coisa com ele também.

**Comment by luizbills** [comment]
Sugiro você obrigar o cadastro antes de poder responder. Assim você pode guardar o progresso do usuário em seu backend, para previnir qualquer trapaça. Sugiro usar backend, pois guardar no localstorage ainda vai permitir ele trapacear.
Se você não quiser ou sobre lidar com backend, você pode usar soluções como o firebase.

**Comment by cezinha** [comment]
seria melhor forçar ele fazer um login para garantir que ele sempre vá continuar o quiz da posição que estava mesmo quando ele quiser tentar burlar (ia sugerir um cookie mas se ele abrir em outro browser ou se ele apagar os cookies, já conseguiria burlar)
